lib/resources/resource_test.go

```diff
@@ -0,0 +1,112 @@
+package resources
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+// Tests that we can get Resource info from a Zabbix API call
+func TestGetInfo(t *testing.T) {
+	expected := Result{
+		{
+			HostID:      "10261",
+			Host:        "dummy-host",
+			Name:        "dummy-host",
+			Description: "",
+		},
+	}
+	// mock api call
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		res := struct {
+			Result `json:"result"`
+		}{expected}
+		err := json.NewEncoder(w).Encode(res)
+		if err != nil {
+			t.Fatalf("unable to write response. error: %v", err)
+		}
+	}))
+	defer ts.Close()
+
+	var z Zabbix
+	res, err := z.getInfo("ZABBIX_KEY", ts.URL)
+	if err != nil {
+		t.Fatalf("process ran with err %v, want result to be %v", err, expected)
+	}
+
+	if !(reflect.DeepEqual(res, expected)) {
+		t.Errorf("expected result to be %v", expected)
+	}
+}
+
+// Test that we can get Rundeck resource model document in Yaml format
+// with hosts' information from Zabbix
+func TestMakeResource(t *testing.T) {
+	// mock api call
+	expected := Result{
+		{
+			HostID:      "10261",
+			Host:        "dummy-host",
+			Name:        "dummy-host",
+			Description: "",
+		},
+	}
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		res := struct {
+			Result `json:"result"`
+		}{expected}
+		err := json.NewEncoder(w).Encode(res)
+		if err != nil {
+			t.Fatalf("unable to write response. error: %v", err)
+		}
+	}))
+	defer ts.Close()
+
+	// generate temp file
+	tmpfile, err := ioutil.TempFile("", "testfile.*.yml")
+	if err != nil {
+		t.Fatalf("unable to create temporary file. error: %v", err)
+	}
+
+	defer os.Remove(tmpfile.Name()) // clean up
+
+	var z Zabbix
+	err = z.MakeResource("ZABBIX_KEY", ts.URL, tmpfile.Name())
+	if err != nil {
+		t.Fatalf("expected resource model document to be generated. error: %v", err)
+	}
+
+	// check the file's content
+	e := `
+dummy-host:
+  hostname: dummy-host
+  nodename: dummy-host
+  description: ""
+  osArch: ""
+  osFamily: ""
+  osName: ""
+  osVersion: ""
+  tags: ""
+  username: ""
+  ssh-keypath: ""`
+
+	c, err := ioutil.ReadFile(tmpfile.Name())
```

Comment:
Hmm ... `tmpFile` has already been opened for reading at the point of creating. Instead of adding another open descriptor to it might just use `tmpFile.Read(xxx)...`

Might be difficult given you'll want to know the number of bytes to read 🤔 ... oh well, it's only test but something to bear in mind.